### PR TITLE
New modules + refactoring

### PR DESCRIPTION
**Pull request by takinosaji**
1) HealthCheck.Mvc module: convenient extension with ready to use controller, view for healthcheck page for your MVC application easy configurable and injectable into main application by several lines of code.
2) HealthCheck.WebApi  module: convenient extension with ready to use controller producing json result for your WebApi application easy configurable and injectable into main application by several lines of code.
3) HealthCheckResult is now in failed state if no checks spotted
4) Configurators added
5) CheckerBase was changed. Properties converted to constructor parameters to encourage using of IoC.

### CODE REVIEW

**Review by fjeldstad — COMMENTED** [review]
Hi @takinosaji! 

First of all, big thanks for taking the time to submit a pull request. This project has been dormant for some time but it's always nice to see that people are actually using it.

I will try to find some time to look into your PR in detail in the coming weeks, but until then I have some general comments:

- Consider the impact on other users when you're suggesting breaking changes. I'm thinking specifically about renaming `HealthCheck.Core.Helpers` to `HealthCheck.Core.Extensions`- what value is this rename adding? Is the new name really worth a breaking change?
- Why abstract `CheckResult` into `ICheckResult`? I'm all for programming against interfaces rather than classes but `CheckResult` is a simple DTO. Do you see a scenario where someone would like to override it with some other behavior? Could you give an example?

Again, thanks for the work you're putting into this!

**Review by fjeldstad — CHANGES_REQUESTED** [review]
Please use C# type aliases whenever possible instead of .NET Framework type names - for example `bool` instead of `Boolean`, `int` instead of `Int32` and so forth.

**Review by fjeldstad — CHANGES_REQUESTED** [review]
Typo: the namespace (and corresponding folder) `HealthCheck.Mvc.ViewDescovery` should be `HealthCheck.Mvc.ViewDiscovery`.

**Comment by takinosaji** [comment]
@Hihaj thx for response. 

The goal behind refactoring is making namespaces self explanatory and typical. It's common practice that extension classes are located in Extension folder and sub namespace eventually. Ofc when developing project further we should think about compatibility but i suppose that we should not hesitate to improve existing codebase. Thx nuget helps us moving forward with letter versions of packages and preserving existing versions.

As for C# type aliases i use class names instead of keywords deliberately on my opinion it makes code more consistent and allows to avoid confusion with changing of meaning of certain keywords in next versions of dotnet. Still i wont argue about it and will fix it back as you asking.

As for check result - well i am eager to use IoC and polymorphism as much as i can. In future we can have different types of ICheckResults. For instance HealthCheckResult i thought must be ICheckResult, maybe future SectionChecks could also implement this interface. I agree that it is supposed to be just a DTO and thus it should not have any behavior => interface

**Comment by minorum** [comment]
@Hihaj Hi, would be great to see those changes go live

**Comment by fjeldstad** [comment]
@minorum Which of the changes are you referring to specifically?

**Comment by minorum** [comment]
Specifically the Mvc and WebApi modules

File: src/HealthCheck.Core/HealthCheck.cs [hunk 1]
```diff
@@ -2,55 +2,53 @@
...
                 // checker individually.
-                var checkResults = _checkers
-                    .AsParallel()
+                var checkResults = Checkers.AsParallel()
                     .Select(async x =>
                     {
                         try
                         {
-                            return await x.Check().ConfigureAwait(false);
+                            return !x.PreserveContext ? await x.Check().ConfigureAwait(false) : x.Check().Result;
```
Comment (fjeldstad) [inline src/HealthCheck.Core/HealthCheck.cs]: What's the purpose of `PreserveContext`? When would you set it to `true`? Are you using this or have you added it for future use?

File: src/HealthCheck.Core/HealthCheck.cs [hunk 2]
```diff
@@ -2,55 +2,53 @@
...
                                 Output = ex.Message
                             };
                         }
-                    })
-                    .ToArray();
+                    }).ToArray();
                 await Task.WhenAll(checkResults).ConfigureAwait(false);
-                return new HealthCheckResult(checkResults.Select(x => x.Result));
+
+                var result = new HealthCheckResult(checkResults.Select(x => x.Result));
```
Comment (fjeldstad) [inline src/HealthCheck.Core/HealthCheck.cs]: Unnecessary temporary variable.

File: src/HealthCheck.Core/Results/ICheckResult.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+
+namespace HealthCheck.Core.Results
+{
+    public interface ICheckResult
```
Comment (fjeldstad) [inline src/HealthCheck.Core/Results/ICheckResult.cs]: I don't feel this interface adds any real value. `CheckResult` is just a data container (DTO if you like) - no need to create another abstraction. It has no logic or behavior. What is the use case for the interface? When would you create a new implementation of it? I would like to see it removed until there is a real need for it.

File: src/HealthCheck.Core/Results/HealthCheckResult.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+        private readonly bool _passed;
+
+        public ICheckResult[] Results => _results;
+        public bool Passed => _passed;
+        public string Output => _passed ? "SUCCESS" : "FAILURE";
+
+        public HealthCheckResult(IEnumerable<ICheckResult> results)
+        {
+            _results = results.OrderBy(x => x.CheckerName).ToArray();
+            _passed = _results.Any() && _results.All(x => x.Passed);
```
Comment (fjeldstad) [inline src/HealthCheck.Core/Results/HealthCheckResult.cs]: I don't see the value in inverting the logic of empty healthchecks (no checkers). A health check passes if there are no failures, there should be no requirement for "at least one checker" - alternatively the `HealthCheck` constructor should throw if no checkers are passed to it.

File: src/HealthCheck.Core/Extensions/TaskExtensions.cs [hunk 5]
```diff
@@ -1,7 +1,7 @@
 ﻿using System;
 using System.Threading.Tasks;
 
-namespace HealthCheck.Core.Helpers
+namespace HealthCheck.Core.Extensions
```
Comment (fjeldstad) [inline src/HealthCheck.Core/Extensions/TaskExtensions.cs:4]: Unnecessary namespace name change.

File: src/HealthCheck.Core/HealthCheckBase.cs [hunk 6]
```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using HealthCheck.Core.Results;
+
+namespace HealthCheck.Core
+{
+    public abstract class HealthCheckBase : IHealthCheck
```
Comment (fjeldstad) [inline src/HealthCheck.Core/HealthCheckBase.cs]: `HealthCheck` is the core component of this library - why would it need to be abstracted? What is `IHealthCheck` and `HealthCheckBase` accomplishing?

File: src/HealthCheck.Core/Results/CheckResult.cs [hunk 7]
```diff
@@ -0,0 +1,10 @@
+﻿namespace HealthCheck.Core.Results
+{
+    public class CheckResult : ICheckResult
+    {
+        public string CheckerName { get; set; }
+        public string SectionName { get; set; }
```
Comment (fjeldstad) [inline src/HealthCheck.Core/Results/CheckResult.cs]: `SectionName` is fine with me as long as it's optional.

File: src/HealthCheck.Core/Results/CheckResult.cs [hunk 8]
```diff
@@ -0,0 +1,10 @@
+﻿namespace HealthCheck.Core.Results
+{
+    public class CheckResult : ICheckResult
+    {
+        public string CheckerName { get; set; }
```
Comment (fjeldstad) [inline src/HealthCheck.Core/Results/CheckResult.cs]: Renaming `Name` to `CheckerName` is a pretty rough breaking change. I can see that `CheckerName` could be a better name, but I'm not sure it's _better enough_ to justify breaking all real-world tests.

File: src/HealthCheck.Core/Results/HealthCheckResult.cs [hunk 9]
```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace HealthCheck.Core.Results
```
Comment (fjeldstad) [inline src/HealthCheck.Core/Results/HealthCheckResult.cs]: I would like `HealthCheckResult` to remain in the `HealthCheck.Core` namespace. There won't be many different types of results, so introducing a new namespace feels unnecessary.

File: src/HealthCheck.Mvc/Models/HealthCheckResultModel.cs [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+                TotalChecks = result.Results.Length,
+                PassedChecks = result.Results.Count(r => r.Passed),
+                FailedChecks = result.Results.Count(r => !r.Passed),
+            };
+            Sections = result.Results
+                .Select(r =>
+                {
+                    if (string.IsNullOrEmpty(r.SectionName))
+                    {
+                        r.SectionName = "#";
```
Comment (fjeldstad) [inline src/HealthCheck.Mvc/Models/HealthCheckResultModel.cs:31]: Why "#"?

File: src/HealthCheck.Core/Configuration/IHealthCheckConfigurator.cs [hunk 11]
```diff
@@ -0,0 +1,10 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace HealthCheck.Core.Configuration
+{
+    public interface IHealthCheckConfigurator
+    {
+        IHealthCheckConfigurator ConfigureCheckers(Func<IEnumerable<IChecker>> getChechers);
```
Comment (fjeldstad) [inline src/HealthCheck.Core/Configuration/IHealthCheckConfigurator.cs:8]: Why return `IHealthCheckConfigurator`? Do you see a scenario in which one would chain multiple `ConfigureCheckers(...)` after one another?

**Review by fjeldstad — CHANGES_REQUESTED** [review]
Hi @takinosaji - sorry for the super-late feedback, I've been busy at work.

I've finally put aside some time to review your PR and have left a few comments. Overall I would very much like to have ASP.NET MVC 5 and Web API 2 support in Hihaj.HealthCheck and I appreciate the effort you've put into this.

Looking at your code and comments I feel that we have a bit different approaches to design though; you said earlier that you want to use polymorphism "as much as possible" while I prefer to use it only when necessary. Abstraction creates a mental overhead and makes any codebase harder to use which is something I would like to avoid.

Please have a look at my comments and see if you would consider making some adjustments. Focus on keeping the library simple for the user (consuming code). Also keep breaking changes to an absolute minimum.

File: src/HealthCheck.Core/HealthCheck.cs (hunk 1 above)
Comment (takinosaji) [reply src/HealthCheck.Core/HealthCheck.cs]: Because we have faced situation when we do need capturing context. If i am not mistaking when using it from mvc or web api controller we are loosing web context which used in one particular healthcheck.

File: src/HealthCheck.Core/HealthCheck.cs (hunk 2 above)
Comment (takinosaji) [reply src/HealthCheck.Core/HealthCheck.cs]: fixed

File: src/HealthCheck.Core/Results/ICheckResult.cs (hunk 3 above)
Comment (takinosaji) [reply src/HealthCheck.Core/Results/ICheckResult.cs]: fixed

File: src/HealthCheck.Core/Results/HealthCheckResult.cs (hunk 9 above)
Comment (takinosaji) [reply src/HealthCheck.Core/Results/HealthCheckResult.cs]: frankly speaking i dont consider keeping bunch of files on same folder as a good practice but will not argue with you on this matter. Will change)

File: src/HealthCheck.Core/Extensions/TaskExtensions.cs (hunk 5 above)
Comment (takinosaji) [reply src/HealthCheck.Core/Extensions/TaskExtensions.cs:4]: Having helpers it is like having something bad designed in codebase. I am trying to avoid helper classes and namespaces, use more strict definitions. Second point is that we have extension methods which are special c# feature. We can and will have them more than for one class and it would be nice to keep them separated from other "helpers".

File: src/HealthCheck.Mvc/Models/HealthCheckResultModel.cs (hunk 10 above)
Comment (takinosaji) [reply src/HealthCheck.Mvc/Models/HealthCheckResultModel.cs:31]: If you will look at mvc page of healthcheck you shall admit that having empty title is looking bad. # looks good) It is like "Unnamed section"

File: src/HealthCheck.Core/HealthCheckBase.cs (hunk 6 above)
Comment (takinosaji) [reply src/HealthCheck.Core/HealthCheckBase.cs]: To have different healthcheck in future. Not insisting. Will fix.

File: src/HealthCheck.Core/Results/CheckResult.cs (hunk 8 above)
Comment (takinosaji) [reply src/HealthCheck.Core/Results/CheckResult.cs]: Well, i,d keep it that way. Breaking changes are not avoidable. Checker has SessionName already. Since we have multiple names in one class lets name it as strict as possible

File: src/HealthCheck.Core/Results/CheckResult.cs (hunk 7 above)
Comment (takinosaji) [reply src/HealthCheck.Core/Results/CheckResult.cs]: it is optional if i got your point

File: src/HealthCheck.Core/Configuration/IHealthCheckConfigurator.cs (hunk 11 above)
Comment (takinosaji) [reply src/HealthCheck.Core/Configuration/IHealthCheckConfigurator.cs:8]: Yep. When we shall have more methods than one for configuration having the opportunity to use call-chain style will be more than convenient.

File: src/HealthCheck.Core/HealthCheck.cs (hunk 1 above)
Comment (fjeldstad) [reply src/HealthCheck.Core/HealthCheck.cs]: I don't think it's a good idea to use for example `HttpContext.Current` in a checker implementation either way, from what I've read it's not designed for being used from different threads... My understanding is that `ConfigureAwait(false)` is used to tell the runtime that the _calling_ code (which calls `await someTask`) does not need to be returned to the same context after the `await` - not that `someTask` needs the original context.

File: src/HealthCheck.Core/Extensions/TaskExtensions.cs (hunk 5 above)
Comment (fjeldstad) [reply src/HealthCheck.Core/Extensions/TaskExtensions.cs:4]: "Helpers" is just a name, I don't think it's indicative of bad design in itself.

File: src/HealthCheck.Core/Configuration/IHealthCheckConfigurator.cs (hunk 11 above)
Comment (fjeldstad) [reply src/HealthCheck.Core/Configuration/IHealthCheckConfigurator.cs:8]: Then let's add that feature when it's needed instead. You can't _know_ that there will ever be more configuration methods, right?

File: src/HealthCheck.Core/CheckerBase.cs [hunk 12]
```diff
@@ -1,26 +1,42 @@
...
 
 namespace HealthCheck.Core
 {
     public abstract class CheckerBase : IChecker
     {
-        private TimeSpan? _timeout = TimeSpan.FromSeconds(5);
+        public string Name { get; set; }
+        public string SectionName { get; set; }
+        public virtual bool PreserveContext => false;
+        public virtual TimeSpan? Timeout { private get; set; }
```
Comment (fjeldstad) [inline src/HealthCheck.Core/CheckerBase.cs]: Public setter and private getter seems a bit strange, let's make the getter public as well.

File: src/HealthCheck.Core/HealthCheckResult.cs [hunk 13]
```diff
@@ -8,14 +8,14 @@ public class HealthCheckResult
         private readonly CheckResult[] _results;
         private readonly bool _passed;
 
-        public CheckResult[] Results { get { return _results; } }
-        public bool Passed { get { return _passed; } }
-        public string Status { get { return _passed ? "success" : "failure"; } }
+        public CheckResult[] Results => _results;
+        public bool Passed => _passed;
+        public string Output => _passed ? "SUCCESS" : "FAILURE";
```
Comment (fjeldstad) [inline src/HealthCheck.Core/HealthCheckResult.cs]: Why the change from lower case to upper case? If someone has written a test against the healthcheck result, it would break. I don't see the value in changing this.

**Review by fjeldstad — CHANGES_REQUESTED** [review]
The readme would need to be updated with examples of how to configure the new ASP.NET MVC and Web API modules.

File: src/HealthCheck.Core/HealthCheckResult.cs (hunk 13 above)
Comment (fjeldstad) [reply src/HealthCheck.Core/HealthCheckResult.cs]: Also I want the output to be as compatible as possible with [Runscope/healthcheck](https://github.com/Runscope/healthcheck).

File: src/HealthCheck.Core/CheckResult.cs [hunk 14]
```diff
@@ -2,8 +2,9 @@
 {
     public class CheckResult
     {
-        public string Checker { get; set; }
+        public string CheckerName { get; set; }
```
Comment (fjeldstad) [inline src/HealthCheck.Core/CheckResult.cs]: It will be called "Checker" to be compatible with [Runscope/healthcheck](https://github.com/Runscope/healthcheck).

**Review by fjeldstad — CHANGES_REQUESTED** [review]

File: src/HealthCheck.Core/Configuration/IHealthCheckConfigurator.cs (hunk 11 above)
Comment (takinosaji) [reply src/HealthCheck.Core/Configuration/IHealthCheckConfigurator.cs:8]: Basically i am using it currently. I dont call 
configurator.ConfigureCheckers(..)
configurator.ConfigureCheckers(..)

i use 

```
configurator
            .ConfigureCheckers(..)
            .ConfigureCheckers(..)
```

looks much better

File: src/HealthCheck.Core/HealthCheck.cs (hunk 1 above)
Comment (takinosaji) [reply src/HealthCheck.Core/HealthCheck.cs]: Well... you have created universal tool for implementing healthchecks. Healthcheck term is so abstract it can be anything. Even value from current web context for particular project. You cannot restrict people in capabilities of your tool - they just wont use it right?
Do you want concrete example? - We have been implementing some healhcheck for umbraco based project. For reusing some of built-in healthchecks available only in ui for our web api call we NEEDED current context.

File: src/HealthCheck.Core/HealthCheckResult.cs (hunk 13 above)
Comment (takinosaji) [reply src/HealthCheck.Core/HealthCheckResult.cs]: well.. it is just looks better in mvc page out of the box. I have added mvc module in framework whoch shows your healthcheck in simple page. So it's just for cosmetic look. but ok

File: src/HealthCheck.Core/CheckResult.cs (hunk 14 above)
Comment (takinosaji) [reply src/HealthCheck.Core/CheckResult.cs]: Do you want me to return Checker? But it is not checker... it is his only name. Why should we copy flaws of other projects?

File: src/HealthCheck.Core/Extensions/TaskExtensions.cs (hunk 5 above)
Comment (takinosaji) [reply src/HealthCheck.Core/Extensions/TaskExtensions.cs:4]: Well it s not just a name. It is a part of solution structure. It is something we are working with on daily basis and on which we are relying on. We want to look at collapsed folder name and not to play quiz what is inside. It is very clear that only extension method classes will be located in that folder <=> namespace

File: src/HealthCheck.Core/HealthCheckResult.cs (hunk 13 above)
Comment (fjeldstad) [reply src/HealthCheck.Core/HealthCheckResult.cs]: If it's for cosmetic reasons, I suggest using CSS to make the status value uppercase. That way it only affects the MVC view and not all (existing) endpoints.

File: src/HealthCheck.Core/Configuration/IHealthCheckConfigurator.cs (hunk 11 above)
Comment (fjeldstad) [reply src/HealthCheck.Core/Configuration/IHealthCheckConfigurator.cs:8]: Ok but why are you setting up your checkers in two phases in the first place? Why not just pass in all checkers in the first call?

File: src/HealthCheck.Mvc/Models/HealthCheckResultModel.cs (hunk 10 above)
Comment (fjeldstad) [reply src/HealthCheck.Mvc/Models/HealthCheckResultModel.cs:31]: Ok. Would have preferred that to be part of the view logic rather than the view model in the case the client has requested for example a JSON version of the endpoint to use in automatic tests.

File: src/HealthCheck.Core/Configuration/IHealthCheckConfigurator.cs (hunk 11 above)
Comment (takinosaji) [reply src/HealthCheck.Core/Configuration/IHealthCheckConfigurator.cs:8]: It just looks more vivid and descriptive from code base. Imagine you have 25 checkers which logically may be divided into 5 groups. I'd separate them this way in code to obtain more readability. It works)

File: src/HealthCheck.Core/HealthCheck.cs (hunk 1 above)
Comment (fjeldstad) [reply src/HealthCheck.Core/HealthCheck.cs]: Ok then.

Could you change that line into this though:
```
return !x.PreserveContext ? await x.Check().ConfigureAwait(false) : await x.Check();
```
or even
```
return await x.Check().ConfigureAwait(x.PreserveContext);
```

File: src/HealthCheck.Core/Extensions/TaskExtensions.cs (hunk 5 above)
Comment (fjeldstad) [reply src/HealthCheck.Core/Extensions/TaskExtensions.cs:4]: I actually think "Helpers" is a better name than "Extensions" in this case; it's signalling that this subfolder contains utility components used internally by the "Core" components. Calling it "Extensions" would lead one to think that there are only extension methods inside, or that the components are extensions to "Core" itself (similar to the convention of naming NuGet "addon" packages to [main package].Extensions, for example `Nlog.Extensions`). "Utilities" would also be a good name, but I don't see the point in changing it.

File: src/HealthCheck.Core/HealthCheck.cs (hunk 1 above)
Comment (takinosaji) [reply src/HealthCheck.Core/HealthCheck.cs]: I suppose i have confused you a bit. As far as i remember (it was 6 month ago) we exactly needed to run checker at the same thread to have an access to WebContext. That s why i "awaited" task but asked for .Result. I made a change you required i suppose guys will report here if it will be an issue again. 

@minorum  Yuo should check it guys in your projects. If your healthcheck will become broken we will need to redesign this change back.

File: src/HealthCheck.Core/CheckResult.cs (hunk 14 above)
Comment (fjeldstad) [reply src/HealthCheck.Core/CheckResult.cs]: To the caller of the healthcheck endpoint there is no distinction between a "checker" and its name. The checker instance is an internal object and the name is the only public reference to the checker. Therefore it makes sense to just call it "checker". I agree that _internally_ "checker" is a different concept (which contains a "name" property) but that's just not important to the user.

File: src/HealthCheck.Core/HealthCheck.cs (hunk 1 above)
Comment (fjeldstad) [reply src/HealthCheck.Core/HealthCheck.cs]: I see...it still seems a bit strange to me though; the `HttpContext.Current` reflects the current request, which in all cases for the healthcheck endpoint would probably be initiated by some kind of monitoring or automatic test system. What request-dependant properties are you performing health checks on? (I'm not saying that it's wrong, I'm just curious...)

File: src/HealthCheck.Mvc/MvcHealthCheckСonfigurator.cs [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+using System.Web.Routing;
+using HealthCheck.Core;
+using HealthCheck.Core.Configuration;
+using HealthCheck.Mvc.ViewDiscovery;
+
+namespace HealthCheck.Mvc
+{
+    public class MvcHealthCheckСonfigurator : ConfiguratorBase
+    {
+        private static MvcHealthCheckСonfigurator _instance;
```
Comment (fjeldstad) [inline src/HealthCheck.Mvc/MvcHealthCheckСonfigurator.cs:15]: Make this `private static readonly` and create instance on same row. No need to wait until `GetInstance()` is called since there is virtually zero overhead in creating the instance.

File: src/HealthCheck.Mvc/Controllers/MvcHealthCheckController.cs [hunk 16]
```diff
@@ -0,0 +1,34 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.Remoting.Messaging;
+using System.Web.Mvc;
+using HealthCheck.Core;
+using HealthCheck.Mvc.Models;
+
+namespace HealthCheck.Mvc.Controllers
+{
+    public class MvcHealthCheckController : AsyncController
```
Comment (fjeldstad) [inline src/HealthCheck.Mvc/Controllers/MvcHealthCheckController.cs:11]: Don't inherit from `AsyncController`, instead implement async controller method as described in [the ASP.NET MVC 4 docs](https://docs.microsoft.com/en-us/aspnet/mvc/overview/performance/using-asynchronous-methods-in-aspnet-mvc-4) (using `Task<ActionResult>`).

File: src/HealthCheck.Core/Extensions/TaskExtensions.cs (hunk 5 above)
Comment (takinosaji) [reply src/HealthCheck.Core/Extensions/TaskExtensions.cs:4]: Helpers may be parent folder for Extensions folder but i strongly disagree on mixing up this and having such general parts of project. Again, if you so strongly disagree after this comment i will change it back

File: src/HealthCheck.Mvc/Controllers/MvcHealthCheckController.cs [hunk 17]
```diff
@@ -0,0 +1,34 @@
...
+        {
+            AsyncManager.OutstandingOperations.Increment();
+            AsyncManager.Parameters[Key] = _healthCheck.Run().Result;
+            AsyncManager.OutstandingOperations.Decrement();
+        }
+
+        public ActionResult RunCompleted(HealthCheckResult result)
+        {            
+            var model = new HealthCheckResultModel(result);
+            return View("~/Views/HealthCheck.cshtml", model);
```
Comment (fjeldstad) [inline src/HealthCheck.Mvc/Controllers/MvcHealthCheckController.cs]: Use the built-in content negotiation to return the requested view instead of always forcing the response to be HTML. So just use `return View(model);` - I'm not sure if the view locator would need some additional hint to find the correct HTML view in this case but I think it will work out of the box.

This way, users can get a JSON healthcheck endpoint in an ASP.NET MVC project just by setting the `Accept` header when requesting it.

File: src/HealthCheck.Mvc/Controllers/MvcHealthCheckController.cs [hunk 18]
```diff
@@ -0,0 +1,34 @@
...
+        public void RunAsync()
+        {
+            AsyncManager.OutstandingOperations.Increment();
+            AsyncManager.Parameters[Key] = _healthCheck.Run().Result;
+            AsyncManager.OutstandingOperations.Decrement();
+        }
+
+        public ActionResult RunCompleted(HealthCheckResult result)
+        {            
+            var model = new HealthCheckResultModel(result);
```
Comment (fjeldstad) [inline src/HealthCheck.Mvc/Controllers/MvcHealthCheckController.cs]: ...also I would really prefer if you used the `HealthCheckResult` instance directly as view model instead of this new `HealthCheckResultModel` - that would make the "raw" output from the healthcheck endpoint look the same regardless of whether one is using the Nancy module or the MVC one. (You can still format it differently in the HTML view.)

File: src/HealthCheck.Mvc/Controllers/MvcHealthCheckController.cs (hunk 17 above)
Comment (takinosaji) [reply src/HealthCheck.Mvc/Controllers/MvcHealthCheckController.cs]: Actually i didnt get the point. Lack of experience here... could you please elaborate?

File: src/HealthCheck.Mvc/MvcHealthCheckСonfigurator.cs (hunk 15 above)
Comment (takinosaji) [reply src/HealthCheck.Mvc/MvcHealthCheckСonfigurator.cs:15]: it is used outside to make it private

File: src/HealthCheck.Mvc/Controllers/MvcHealthCheckController.cs (hunk 18 above)
Comment (takinosaji) [reply src/HealthCheck.Mvc/Controllers/MvcHealthCheckController.cs]: ok

File: src/HealthCheck.Mvc/Controllers/MvcHealthCheckController.cs (hunk 17 above)
Comment (takinosaji) [reply src/HealthCheck.Mvc/Controllers/MvcHealthCheckController.cs]: I have already added additional virtual path provider to discover that view. Dont remember exactly but it was not easy or even possible to use what are you talking about.
